### ISSUE TITLE
[Question] Dynamic constraints/Removing constraints

### ISSUE THREAD

**Issue by theSherwood** [lume/autolayout#3]
Looking at the API, there don't seem to be any options for removing constraints or subviews. I didn't see anything in the docs or examples about how to handle the case of subviews getting removed or their relationships to other subviews changing. Is there any way to do that with this library?

**Comment by trusktr** [comment]
Currently we just make a new auto layout view with the new constraints like so:

https://github.com/lume/lume/blob/deb0d9a90db442dd4793e6e0920a252c31c5cce6/src/layouts/Autolayout.ts#L149-L152

**Comment by trusktr** [comment]
For now that's the only official way, as the `View` only has `addConstraint()` and no reciprocal `removeConstraint()`:

- https://github.com/lume/autolayout/blob/6d4f77a0cca9be8526a185ebeb85816e09e201fa/src/View.ts#L233

The underlying `kiwi.Solver` does have both add and remove methods though:

- https://github.com/lume/kiwi/blob/26e8fdfe464eba3123971c17212a503f740262a2/src/solver.ts#L48
- https://github.com/lume/kiwi/blob/26e8fdfe464eba3123971c17212a503f740262a2/src/solver.ts#L105

To optimize this so that we don't have to recreate the whole View to change constraints, we need to hook up a new removeConstraints method to kiwi.Solver's.

**Comment by theSherwood** [comment]
Hey, thanks for the quick reply.

> The underlying kiwi.Solver does have both add and remove methods though

I noticed that about the kiwi solver. Is adding `removeConstraint` on the roadmap or are you saying that you'd be open to the right PR?

I guess another wrinkle might be making sure that the VFL/EVFL parsers return the constraints they create so that they can be removed. It would also be nice to be able to remove a SubView and have it automatically remove all associated constraints.

**Comment by theSherwood** [comment]
> I guess another wrinkle might be making sure that the VFL/EVFL parsers return the constraints they create so that they can be removed.

Looks like this is already the case.

I notice that there are also some constraints created implicitly in SubView. Those may or may not be the kinds of constraints that a user might want to remove. I don't know.

**Comment by trusktr** [comment]
Yeah, that would be robust. Add or remove things and it updates as we'd expect.

I don't have the time at the moment to get to this, but would be happy to merge a PR!

**Comment by theSherwood** [comment]
Looking more at this, I think it might make some sense to make some changes to Kiwi first. Are you open to PRs? There's no public API in Kiwi to return the constraints, for example.

**Comment by trusktr** [comment]
Yeah, totally open to PRs. Would be great to add that.
